File: src/main/java/com/dersaun/apigestaocontas/domain/services/ResumoFaturaService.java
package com.dersaun.apigestaocontas.domain.services;

import com.dersaun.apigestaocontas.domain.Pessoa;
import com.dersaun.apigestaocontas.domain.ResumoFatura;
import com.dersaun.apigestaocontas.domain.dtos.MesAnoChartDTO;
import com.dersaun.apigestaocontas.domain.dtos.MesAnoDTO;
import com.dersaun.apigestaocontas.domain.dtos.ResumoFaturaPessoas;
import com.dersaun.apigestaocontas.domain.dtos.ResumoFormaPagamentoMensalResponseDTO;
import org.eclipse.collections.api.list.MutableList;

import java.util.List;

public interface ResumoFaturaService {

    public final long DINHEIRO = 7L;
    public final long CARNE = 6L;

    ResumoFormaPagamentoMensalResponseDTO buscarResumoPagamentos(MesAnoDTO mesAnoDTO, Long idPessoa);

    ResumoFormaPagamentoMensalResponseDTO buscarResumoPagamentosMesAtual(Long idPessoa);

    List<ResumoFaturaPessoas> resumoFaturaPessoas(MesAnoDTO mesAnoDTO);

    MutableList<ResumoFatura> buscarResumoFaturasPorMesAnos(MutableList<? extends MesAnoChartDTO> meses, Pessoa pessoa);
}

File: src/app/conta/interfaces/resumo-fatura.service.ts
import { UsuarioAutenticadoDto } from '@app/autenticacao/dtos/usuario-autenticado.dto';
import { MesAnoChartDTO } from '@app/conta/dtos/mes-ano-chart.dto';
import { MesAnoDTO } from '@app/conta/dtos/mes-ano.dto';
import { ResumoFaturaPessoas } from '@app/conta/dtos/resumo-faturas';
import { ResumoFormaPagamentoMensalResponseDTO } from '@app/conta/dtos/resumo-forma-pagamento-mensal-response.dto';
import { ResumoFatura } from '@app/conta/models/resumo-fatura.entity';
import { Pessoa } from '@app/pessoa/models/pessoa.entity';

export interface IResumoFaturaService {
  buscarResumoPagamentos(
    mesAnoDTO: MesAnoDTO,
    idPessoa: number,
  ): Promise<ResumoFormaPagamentoMensalResponseDTO>;

  buscarResumoPagamentosMesAtual(
    idPessoa: number,
  ): Promise<ResumoFormaPagamentoMensalResponseDTO>;

  resumoFaturaPessoas(
    mesAnoDTO: MesAnoDTO,
    usuario: UsuarioAutenticadoDto,
  ): Promise<ResumoFaturaPessoas[]>;

  buscarResumoFaturasPorMesAnos(
    meses: MesAnoChartDTO[],
    pessoa: Pessoa,
  ): Promise<ResumoFatura[]>;
}
